tensorflow_transform/tf_utils.py

```diff
@@ -146,12 +146,35 @@ def reduce_batch_count(x, reduce_instance_dims):
           dense_shape=x.dense_shape)
       return tf.sparse_reduce_sum(ones_like, axis=0)
 
-  if reduce_instance_dims:
-    return tf.size(x)
+  # Count the non-nan values
+  # If not reduce_instance_dims, count along the batch dim axis, and return a tensor shaped like x
+  axis = None if reduce_instance_dims else 0
+  ones_like = tf.where(tf.is_nan(tf.cast(x, tf.float32)), tf.zeros_like(x), tf.ones_like(x))
```

Comment:
Can we call reduce_batch_sum() here and in the sparse case as well with tf.ones_like/sparse ones like?

Reply:
Made this change in the most recent push